### ISSUE TITLE
endless sasl auth failure stream

### ISSUE THREAD

**Issue by enckse** [oragono/oragono#119]
was playing with authenticate
```
/quote AUTHENTICATE user
```

seems to end up with an endless stream of (repeating every 2-4 seconds or so)
```
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
* SASL authentication failed
```

**Comment by DanielOaks** [comment]
Hmm, this looks strange. Any chance you'd be able to try enabling USERINPUT and USEROUTPUT and send me the log file of what's going on here? I'm interested to see exactly what's being sent by the client (and whether for instance it keeps on trying to send creds).

Usually when doing SASL the first step is to do `AUTHENTICATE PLAIN` or `AUTHENTICATE CERTFP` and go from there... I think a lot of clients automagically respond to SASL numerics through using their own inbuilt handling, maybe it sees the `AUTHENTICATE` response and then starts sending its' own traffic.

If you'd like to throw me those logs personally, feel free to email them my way at daniel@danieloaks.net – in addition, mind throwing me the client you're using? Thanks!

**Comment by enckse** [comment]
enabled (well..removed exclusions for) user in/output,

the command I ran (yes this exact text)
```
/quote AUTHENTICATE user
```

this was immediately after the login to the server completed that I ran the request the PING indicates the end of login and basically when I issued the request
```
2017-06-30T00:34:32Z : debug : userinput  : enckse : <-  : PING LAG1498732871612
2017-06-30T00:34:32Z : debug : useroutput : enckse :  -> : :oragono.server PONG LAG1498732871612
2017-06-30T00:34:41Z : debug : userinput  : enckse : <-  : AUTHENTICATE user
2017-06-30T00:34:41Z : debug : useroutput : enckse :  -> : :oragono.server 904 enckse :SASL authentication failed
2017-06-30T00:34:41Z : debug : userinput  : enckse : <-  : AUTHENTICATE *
2017-06-30T00:34:41Z : debug : useroutput : enckse :  -> : :oragono.server 904 enckse :SASL authentication failed
2017-06-30T00:34:41Z : debug : userinput  : enckse : <-  : AUTHENTICATE *
2017-06-30T00:34:41Z : debug : useroutput : enckse :  -> : :oragono.server 904 enckse :SASL authentication failed
2017-06-30T00:34:41Z : debug : userinput  : enckse : <-  : AUTHENTICATE *
2017-06-30T00:34:41Z : debug : useroutput : enckse :  -> : :oragono.server 904 enckse :SASL authentication failed
2017-06-30T00:34:41Z : debug : userinput  : enckse : <-  : AUTHENTICATE *
2017-06-30T00:34:41Z : debug : useroutput : enckse :  -> : :oragono.server 904 enckse :SASL authentication failed
2017-06-30T00:34:42Z : debug : userinput  : enckse : <-  : AUTHENTICATE *
2017-06-30T00:34:42Z : debug : useroutput : enckse :  -> : :oragono.server 904 enckse :SASL authentication failed
2017-06-30T00:34:43Z : debug : userinput  : enckse : <-  : AUTHENTICATE *
2017-06-30T00:34:43Z : debug : useroutput : enckse :  -> : :oragono.server 904 enckse :SASL authentication failed
2017-06-30T00:34:46Z : debug : userinput  : enckse : <-  : AUTHENTICATE *
2017-06-30T00:34:46Z : debug : useroutput : enckse :  -> : :oragono.server 904 enckse :SASL authentication failed
2017-06-30T00:34:47Z : debug : userinput  : enckse : <-  : AUTHENTICATE *
2017-06-30T00:34:47Z : debug : useroutput : enckse :  -> : :oragono.server 904 enckse :SASL authentication failed
2017-06-30T00:34:49Z : debug : userinput  : enckse : <-  : AUTHENTICATE *
2017-06-30T00:34:49Z : debug : useroutput : enckse :  -> : :oragono.server 904 enckse :SASL authentication failed
2017-06-30T00:34:51Z : debug : userinput  : enckse : <-  : AUTHENTICATE *
2017-06-30T00:34:51Z : debug : useroutput : enckse :  -> : :oragono.server 904 enckse :SASL authentication failed
2017-06-30T00:34:53Z : debug : userinput  : enckse : <-  : AUTHENTICATE *
2017-06-30T00:34:53Z : debug : useroutput : enckse :  -> : :oragono.server 904 enckse :SASL authentication failed
2017-06-30T00:34:55Z : debug : userinput  : enckse : <-  : AUTHENTICATE *
2017-06-30T00:34:55Z : debug : useroutput : enckse :  -> : :oragono.server 904 enckse :SASL authentication failed
2017-06-30T00:34:57Z : debug : userinput  : enckse : <-  : AUTHENTICATE *
2017-06-30T00:34:57Z : debug : useroutput : enckse :  -> : :oragono.server 904 enckse :SASL authentication failed
2017-06-30T00:34:59Z : debug : userinput  : enckse : <-  : AUTHENTICATE *
2017-06-30T00:34:59Z : debug : useroutput : enckse :  -> : :oragono.server 904 enckse :SASL authentication failed
2017-06-30T00:35:01Z : debug : userinput  : enckse : <-  : AUTHENTICATE *
2017-06-30T00:35:01Z : debug : useroutput : enckse :  -> : :oragono.server 904 enckse :SASL authentication failed
2017-06-30T00:35:03Z : debug : userinput  : enckse : <-  : AUTHENTICATE *
2017-06-30T00:35:03Z : debug : useroutput : enckse :  -> : :oragono.server 904 enckse :SASL authentication failed
```

I'm definitely capable of _not_ doing stupid stuff so chase this as you have time - it isn't blocking me just appears to an infinite error condition

**Comment by enckse** [comment]
also - I did this from hexchat

**Comment by DanielOaks** [comment]
> The client can abort an authentication by sending an asterisk as the data. The server will send a 906 numeric.

From the IRCv3 SASL docs... we don't do this. This is a problem.

**Comment by DanielOaks** [comment]
This should be resolved as of a few weeks ago, but I'll close this once the fix is in a proper release.

**Comment by DanielOaks** [comment]
I'll close this for now, the fix is in v0.9.0. Feel free to re-open if you keep running into the same issue!
